src-v2/NTS.Persistance/Adapters/SetupLapRepository.cs

```diff
@@ -0,0 +1,44 @@
+﻿using NTS.Domain.Setup.Entities;
+using NTS.Persistence.Setup;
+
+namespace NTS.Persistence.Adapters;
+
+public class SetupLapRepository : SetRepository<Lap, SetupState>
+{
+    public SetupLapRepository(IStore<SetupState> store) : base(store)
+    {
+        state = store.Load().Result;
+    }
+
+    private SetupState state { get; set; }
+    public override Task<Lap> Update(Lap entity)
+    {
+        state.Event.Competitions
+            .Where(competition => competition.Phases.Any(phase => phase.SelectedLap.Id == entity.Id))
```

Comment:
You can use `phase.SelectedLap == entity` due to `Equals` and `GetHashCode` override sin `DomainEntity.